### PR TITLE
FeastExampleGen first iteration

### PR DESCRIPTION
**Pull request by casassg**
Initial stab at implementing FeastExampleGen

- Uses Feast native serialization into yaml files to store RepoConfig
- Maps RetrievalJob to BigQueryRetrievalJob which has a way to extract SQL query needed for feast.
- Converter class lineage to support multiple input categories (for now only BQ) to transform from input to bytes needed to be stored in TFRecord
- Flexible output data format (for now tf.example from BQ only, but flexibility to append more formats as SequenceExample).
- Mostly reuses as much code as possible from base library to avoid having to maintain a large piece of code.

Cleaned folder a bit

### CODE REVIEW

**Comment by casassg** [comment]
CI is failing due to https://github.com/feast-dev/feast/issues/2035

**Comment by casassg** [comment]
Had to fix some stuff in https://github.com/tensorflow/tfx-addons/pull/93

File: tfx_addons/xgboost_evaluator/xgboost_predict_extractor.py [hunk 1]
```diff
@@ -22,9 +22,16 @@
 import numpy as np
```
Comment (BACtaki) [inline tfx_addons/xgboost_evaluator/xgboost_predict_extractor.py]: @casassg Why do we need these changes (`xgboost_evaluator` and `xgboost_evaluator_test`)?

Comment (casassg) [reply tfx_addons/xgboost_evaluator/xgboost_predict_extractor.py]: landed upstream fixes here. This reduces the PR to only feast changes

**Pull request merged**